core/src/com/unciv/ui/victoryscreen/VictoryScreen.kt

```diff
@@ -175,22 +178,84 @@ class VictoryScreen(val worldScreen: WorldScreen) : PickerScreen() {
 
     private fun setCivRankingsTable() {
         val majorCivs = gameInfo.civilizations.filter { it.isMajorCiv() }
-        val civRankingsTable = Table().apply { defaults().pad(5f) }
+
+        contentsTable.clear()
+        
+        if (UncivGame.Current.settings.useDemographics) contentsTable.add(buildDemographicsTable(majorCivs))
+        else contentsTable.add(buildRankingsTable(majorCivs))
+    }
+
+    enum class RankLabels { Demographic, Rank, Value, Best, Average, Worst}
+    private fun buildDemographicsTable(majorCivs: List<CivilizationInfo>): Table {
+        val demographicsTable = Table().apply { defaults().pad(5f) }
+        for (rankLabel in RankLabels.values())   {
+            if (rankLabel == RankLabels.Demographic) {
+                val demoLabel = Table().apply { defaults().pad(5f) }
+                demoLabel.add(rankLabel.name.toLabel()).row()
+                demoLabel.addSeparator().fillX()
+                demographicsTable.add(demoLabel)
+                for (category in RankingType.values()) {
+                    val headers = Table().apply { defaults().pad(5f) }
+                    headers.add(category.name.replace('_', ' ').toLabel()).row()
+                    headers.addSeparator().fillX()
+                    demographicsTable.add(headers)
+                    }
+            } else {
+                demographicsTable.row()
+                demographicsTable.add(rankLabel.name.toLabel())
+
+                for (category in RankingType.values()) {
+                    @Suppress("NON_EXHAUSTIVE_WHEN") // RankLabels.Demographic treated above
+                    when (rankLabel) {
+                        RankLabels.Rank -> { // finds the current player's rank in each category
+                            val rank = majorCivs.filter{ it.isAlive() }.sortedByDescending { it.getStatForRanking(category) }.indexOfFirst { it.civName == gameInfo.currentPlayerCiv.civName } + 1
+                            demographicsTable.add(rank.toLabel())
+                        }
+
+                        RankLabels.Value -> { // finds the current player's value in each category
+                            demographicsTable.add(getCivGroup(gameInfo.currentPlayerCiv, ": " + gameInfo.currentPlayerCiv.getStatForRanking(category).toString(), playerCivInfo)).fillX()
```

Comment:
local `fun Table.addRankCivGroup(civ: CivInfo) = add(getCivGroup(civ, ": " + civ.getStatForRanking(category).toString(), playerCivInfo)).fillX()` such that it sees `category` - define inside for - and reuse for 3 of the 5 rows? Maybe, could shorten the code lines here?

Reply:
Wait no getCivGroup is already class-private and seems to be always called with that ": "+ part, even in the older buildRankingsTable, so it could just learn to do some of the repeated calling pattern. Maybe. Would need to see the whole thing in Studio, too lazy.

Reply:
Added local function addRankCivGroup which is reused to grab and format civ data.